### PR TITLE
Do not callback one SM EVENT_ERROR twice.

### PR DESCRIPTION
**Pull request by oknet**
After #947 (c1ac5f) and #1522 (a128d5) , the EVENT_ERROR which caused by
EPOLLERR will be sent to read.vio._cont first and then write.vio._cont.

The reader SM could close or shutdown(WRITE) the VC, but we do not check
these operations before we callback write.vio._cont. The SM would
received EVENT_ERROR twice if write.vio._cont == read.vio._cont.

### CODE REVIEW

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/143/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/16/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1607/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/17/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/30/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/1712/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/275/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/17/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1608/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/18/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/31/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/1713/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/19/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/32/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/18/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/20/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/33/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/144/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1714/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1609/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1715/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/19/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/145/

**Comment by oknet** [comment]
This is a fix for #1531 to avoid enforce the handler to handle read error and write error.

@scw00 Can you have a try ?

@zwoop May I ignore the AU check failed ?

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1610/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/276/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/277/

**Comment by scw00** [comment]
@oknet  I it work well on my test env, and jtest can not re-product any more !

**Comment by zwoop** [comment]
@oknet Yes, ignore AU tests for now.

**Comment by zwoop** [comment]
Nice! Testing this PR on docs.trafficserver now.

File: iocore/net/UnixNetVConnection.cc [hunk 1]
```diff
@@ -271,9 +271,22 @@ read_from_net(NetHandler *nh, UnixNetVConnection *vc, EThread *thread)
...
     // if it is a non-temporary error, we should die appropriately
     if (err && err != EAGAIN && err != EINTR) {
-      read_signal_error(nh, vc, err);
+      Continuation *reader_cont = vc->read.vio._cont;
+
+      if (read_signal_error(nh, vc, err) == EVENT_DONE) {
+        return;
+      }
+      // If vc is closed or shutdown(WRITE) in last read_signal_error callback,
+      //   or reader_cont is same as write.vio._cont.
```
Comment (zwoop) [inline iocore/net/UnixNetVConnection.cc]: Great comment, this really helps explaining the issue / solution.

**Pull request merged**